explainability/explainability-core/src/main/java/org/kie/kogito/explainability/utils/DataUtils.java

```
@@ -449,16 +453,44 @@ private static void linearizeFeature(List<Feature> flattenedFeatures, Feature f)
-        try (OutputStream outputStream = Files.newOutputStream(path)) {
+        try (Writer writer = Files.newBufferedWriter(path)) {
             List<Value> xAxis = partialDependenceGraph.getX();
             List<Value> yAxis = partialDependenceGraph.getY();
-            outputStream.write("feature,output\n".getBytes(StandardCharsets.UTF_8));
+            CSVFormat format = CSVFormat.DEFAULT.withHeader(
+                    partialDependenceGraph.getFeature().getName(), partialDependenceGraph.getOutput().getName());
+            CSVPrinter printer = new CSVPrinter(writer, format);
             for (int i = 0; i < xAxis.size(); i++) {
-                String line = xAxis.get(i).asString().replace(",", "") + ',' +
-                        yAxis.get(i).asString().replace(",", "") + '\n';
-                outputStream.write(line.getBytes(StandardCharsets.UTF_8));
+                printer.printRecord(xAxis.get(i).asString(), yAxis.get(i).asString());
             }
-            outputStream.flush();
         }
     }
+
+    /**
+     * Read a CSV file into a {@link DataDistribution} object.
+     *
+     * @param file the path to the CSV file
+     * @param schema an ordered list of {@link Type}s as the 'schema', used to determine
+     *        the {@link Type} of each feature / column
+     * @return the parsed CSV as a {@link DataDistribution}
+     * @throws IOException when failing at reading the CSV file
+     */
+    public static DataDistribution readCSV(Path file, List<Type> schema) throws IOException {
+        List<PredictionInput> inputs = new ArrayList<>();
+        try (BufferedReader reader = Files.newBufferedReader(file)) {
+            Iterable<CSVRecord> records = CSVFormat.RFC4180.withFirstRecordAsHeader().parse(reader);
+            for (CSVRecord record : records) {
+                int size = record.size();
+                if (schema.size() == size) {
```

nitpick... should we raise an exception (with a message) if `schema.size() != size`?

---

I was thinking more to be failsafe in case there're malformed rows, but it's ok to be stricter and raise an exception too